service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -163,4 +166,64 @@ public ConnectionFactory registrationServiceConnectionFactory() {
     public HonoClient registrationServiceClient() {
         return new HonoClientImpl(vertx(), registrationServiceConnectionFactory());
     }
+
+    /**
+     * Exposes configuration properties for accessing the credentials service as a Spring bean.
+     * <p>
+     * Sets the <em>amqpHostname</em> to {@code hono-device-registry} if not set explicitly (reflecting that Hono is
+     * providing the credentials API as part of the device registry component).
+     *
+     * @return The properties.
+     */
+    @Qualifier(CredentialsConstants.CREDENTIALS_ENDPOINT)
+    @ConfigurationProperties(prefix = "hono.credentials")
```

Comment:
I agree regarding the ClientConfigProperties, which will still be created (with default values) if no `HONO_CREDENTIALS_` properties exist.
I made the conditional creation on the `credentialsServiceClient` method only, so that without any  `HONO_CREDENTIALS_` properties the client bean is not created.

This works, but of course a property bean that is not used is still created.

Will move it to the property object a (if I have seen it correctly, this was the solution of the blueprint).